### PR TITLE
Add tsconfig, more linting to `common` package

### PR DESCRIPTION
**Pull request by mqp**
Just gradually turning on more checks when it's easy and fixing up trivial issues.

There are lots of interesting typing problems related to "contracts that have some properties" which I don't know how to solve yet, so I am not attempting to turn on `no-explicit-any` until I am a better Typescript programmer and I understand the best solution to these problems.

### CODE REVIEW

**Comment by mqp** [comment]
@akrolsmir Just letting you know that this is the sort of thing I am up to in my spare time, since you seemed interested in the idea of "turning on more linting stuff".

**Comment by ForrestWeiswolf** [comment]
> There are lots of interesting typing problems related to "contracts that have some properties" which I don't know how to solve yet, so I am not attempting to turn on no-explicit-any until I am a better Typescript programmer and I understand the best solution to these problems.

I've been looking into a few of those, work so far is on https://github.com/ForrestWeiswolf/manifold/tree/remove-any, but I figured I should hold off on making a PR until this is merged.

**Review by jahooma — APPROVED** [review]
Cool.

**Pull request merged**
